src/index.ts

```diff
@@ -40,14 +44,55 @@ export function runInContext(
   const program = parse(code, context)
   if (program) {
     if (theOptions.isNativeRunnable) {
+      let transpiled
+      let sourceMapJson
+      let lastStatementSourceMapJson
       try {
+        const temp = transpile(program, context.contextId)
+        // some issues with formatting and semicolons and tslint so
+        transpiled = temp[0]
+        sourceMapJson = temp[1]
+        lastStatementSourceMapJson = temp[2]
         return Promise.resolve({
           status: 'finished',
-          value: sandboxedEval(generate(transpile(program, context.contextId)))
+          value: sandboxedEval(transpiled)
         } as Result)
       } catch (error) {
-        context.errors.push(new ExceptionError(error, UNKNOWN_LOCATION))
-        return Promise.resolve({ status: 'error' } as Result)
+        if (error instanceof RuntimeSourceError) {
+          context.errors.push(error)
```

Comment:
Just checking, do these errors come from stuff like type checks being injected into the code? I'm not sure what triggers this line.

Reply:
Yes, look at operators.ts, there it throws an InvalidNumberOfArguments RuntimeSourceError.